definition_validator/validator.cc

```diff
@@ -516,71 +519,75 @@ void validateOverriding(const core::Context ctx, const ast::ExpressionPtr &tree,
     for (const auto &mixin : klassData->mixins()) {
         auto superMethod = mixin.data(ctx)->findMethod(ctx, name);
         if (superMethod.exists()) {
-            if (superMethod.data(ctx)->flags.isOverloaded) {
-                ENFORCE(!superMethod.data(ctx)->name.isOverloadName(ctx));
+            auto superMethodData = superMethod.data(ctx);
+            if (superMethodData->flags.isOverloaded) {
+                ENFORCE(!superMethodData->name.isOverloadName(ctx));
                 auto overload = ctx.state.lookupNameUnique(core::UniqueNameKind::Overload, name, 1);
-                superMethod = superMethod.data(ctx)->owner.data(ctx)->findMethod(ctx, overload);
+                superMethod = superMethodData->owner.data(ctx)->findMethod(ctx, overload);
                 ENFORCE(superMethod.exists());
             }
 
             overriddenMethods.emplace_back(superMethod);
         }
     }
 
-    if (overriddenMethods.size() == 0 && method.data(ctx)->flags.isOverride &&
-        !method.data(ctx)->flags.isIncompatibleOverride) {
-        if (auto e = ctx.state.beginError(method.data(ctx)->loc(), core::errors::Resolver::BadMethodOverride)) {
-            e.setHeader("Method `{}` is marked `{}` but does not override anything", method.show(ctx), "override");
+    if (overriddenMethods.empty() && methodData->flags.isOverride && !methodData->flags.isIncompatibleOverride) {
+        if (auto e = ctx.state.beginError(methodLoc, core::errors::Resolver::BadMethodOverride)) {
+            if (methodData->flags.isRewriterSynthesized) {
+                e.setHeader("Method `{}` is marked `{}` but the interface only defines a {} method", methodName,
+                            "override", absl::EndsWith(methodName, "=") ? "reader" : "writer");
```

Comment:
We have a method called `isSetter` in `NameRef.h` that handles special cases (this method will say that `==` is a setter method, but it's not).